pyscriptjs/src/components/pyscript.ts

```diff
@@ -96,10 +96,59 @@ export function initHandlers(interpreter: InterpreterClient) {
 }
 
 /** An always same listeners to reduce RAM and enable future runtime changes via MO */
-const pyScriptListener = async ({ type, currentTarget: el }) => {
+const pyScriptEventHandler = async ({ type, currentTarget: el }) => {
+    try {
+        /*  here, we need to:
+                - resolve the user-provided name to a Python object
+                - determine whether that object is the name of a Callable
+                - Determine the number of parameters of that Callable
+                    - If it's 0, set up listener to call it with no args
+                    - If it's 1, set up listener to call it with the event object
+                    - If it's 2, show error/warning
+        */
+
+        const interpreter = elementInterpreter.get(el);
+
+        // Import create_proxy for later user
+        await interpreter.run(`from pyodide.ffi import create_proxy`);
+
+        // Get the value the user provided for their (hopefully) Callable
+        const userFunctionName = el.getAttribute(`py-${type as string}`);
```

Comment:
how about moving this before any `await` and group all `await` into a single `Promise.all` unless we can actually use a single `await` instead? it feels like 3 unnecessary JS<->WASM round trips instead of just one or oen in parallel ... unless, of course, all this should run sequentially, but being async code, I don't see current approach granting any guard around these operations ... thoughts?

Reply:
Honestly this section is likely to change quite a bit as I actually work through the solution, but it's a good thing to keep in mind.

Reply:
To me it's more some sort of mind-set we don't have around this project (not blaming anyone, it's just what I've found here and there) ... we often run multiple `async` when maybe a single one was needed.

In this particular case, I am also wrong in my assumption (on a second read) because  both other evaluations require the `create_proxy` previously imported ... so how about we use really a single await to have them all?

```js
const { result: [ resolvedFunction, pyCallable ] } = await interpreter.run(`
from pyodide.ffi import create_proxy
[
    create_proxy(eval("""${userFunctionName}"""), roundtrip=False)
    create_proxy(callable)
]
`);
```

wouldn't this work or wouldn't this be better at both timing and simplicity? 🤔

Reply:
Oh for sure it would be better, I've mostly left them separated to be able to throw debugging statements in as I'm trying to track down how our integration of Synclink (or my understanding of it) is somehow being problematic. 

What I'd like to do, of course, is to eliminate the multiple-round-trips like you said:
```js
const pyInspectModule = (await remote_interpreter).pyimport('inspect')
const num_params = pyInspectModule.signature(pyMethod).parameters.length
```
That exact syntax works fine in a [minimal Synclink/Pyodide setup I created for testing](https://github.com/JeffersGlass/minimal-synclink-pyodide/blob/inspect-testing/src/main.ts), but somehow *doesn't* work in this situation... and I'm trying to tease out why. 

If you have a little time and wanted to try and pick this apart with me, I wouldn't mind! There's [simple build instructions](https://github.com/JeffersGlass/minimal-synclink-pyodide) for that other demo if it's helpful.

Reply:
Let's try next week maybe? I was planning to finish early today not just because it's Friday but because I've worked the day I supposed to be off so I am trying to compensate my hours (as contractor) ... hope that makes sense.

I'll have a look regardless and see if I can bring any help there.